File: 0547-number-of-provinces/0547-number-of-provinces.cpp
class Solution {
public:

    bool dfs(vector<vector<int>>& isConnected,int node,vector<int>& visited) {

        if(visited[node]==true) return true;
        
        visited[node]=true;
            for(int x=0;x<isConnected[node].size();x++ ) {
                if(isConnected[node][x] && visited[x]==false) {
                    dfs(isConnected,x,visited);
                }
           
        } 
        return false;
        
    }

    int findCircleNum(vector<vector<int>>& isConnected) {
      vector<int> visited(isConnected.size(),0);
      int nProvince=0;
      for(int i=0;i<isConnected.size();i++) {
          if(visited[i]==0) {
              dfs(isConnected,i,visited);
              nProvince+=1;
          }
      }  
      return nProvince;
    }
};

File: 0547-number-of-provinces/0547-number-of-provinces.py
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        #Adjacency matrix
        visited = set()
        n = len(isConnected)


        def dfs(i):
            if i in visited:
                return
            visited.add(i)
            for neighbour in range(n):
                if neighbour not in visited and isConnected[i][neighbour]==1:
                    dfs(neighbour)
            



        count=0
        for i in range(n):
            if i not in visited:
                dfs(i)
                count+=1
        return count
        